modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditSizeLimitsMVCActionCommand.java

```diff
@@ -70,26 +72,42 @@ protected void doProcessAction(
 
 		long fileMaxSize = ParamUtil.getLong(actionRequest, "fileMaxSize");
 
-		if (scope.equals(
-				ExtendedObjectClassDefinition.Scope.COMPANY.getValue())) {
+		try {
+			if (scope.equals(
+					ExtendedObjectClassDefinition.Scope.COMPANY.getValue())) {
 
-			_dlSizeLimitConfigurationProvider.updateCompanySizeLimit(
-				scopePK, fileMaxSize, _getMimeTypeSizeLimits(actionRequest));
-		}
-		else if (scope.equals(
-					ExtendedObjectClassDefinition.Scope.GROUP.getValue())) {
+				_dlSizeLimitConfigurationProvider.updateCompanySizeLimit(
+					scopePK, fileMaxSize,
+					_getMimeTypeSizeLimits(actionRequest));
+			}
+			else if (scope.equals(
+						ExtendedObjectClassDefinition.Scope.GROUP.getValue())) {
 
-			_dlSizeLimitConfigurationProvider.updateGroupSizeLimit(
-				scopePK, fileMaxSize, _getMimeTypeSizeLimits(actionRequest));
-		}
-		else if (scope.equals(
-					ExtendedObjectClassDefinition.Scope.SYSTEM.getValue())) {
+				_dlSizeLimitConfigurationProvider.updateGroupSizeLimit(
+					scopePK, fileMaxSize,
+					_getMimeTypeSizeLimits(actionRequest));
+			}
+			else if (scope.equals(
+						ExtendedObjectClassDefinition.Scope.SYSTEM.
+							getValue())) {
 
-			_dlSizeLimitConfigurationProvider.updateSystemSizeLimit(
-				fileMaxSize, _getMimeTypeSizeLimits(actionRequest));
+				_dlSizeLimitConfigurationProvider.updateSystemSizeLimit(
+					fileMaxSize, _getMimeTypeSizeLimits(actionRequest));
+			}
+			else {
+				throw new PortalException("Unsupported scope: " + scope);
+			}
 		}
-		else {
-			throw new PortalException("Unsupported scope: " + scope);
+		catch (Exception exception) {
+			if (exception instanceof ConfigurationModelListenerException) {
+				SessionErrors.add(actionRequest, exception.getClass());
+
+				actionResponse.sendRedirect(
+					ParamUtil.getString(actionRequest, "redirect"));
+			}
+			else {
+				throw exception;
+			}
```

Comment:
Same as with my previous comment. Also, here it may be worth extracting the code inside the try/catch to its own private method to make things easier to read.

Reply:
Done! Thanks for your feedback